gtfs/spec/en/reference.md

```diff
@@ -295,6 +305,70 @@ For examples that demonstrate how to specify a fare structure with [fare_rules.t
+
+## fare_transfer_rules.txt
+
+File: **Optional**
+
+Fare rules for transfers between legs of travel defined in `fare_leg_rules.txt`.
+
+Transfer costs in `fare_transfer_rules.txt` are queried by filtering fields until a transfer cost matches the characteristics of the transfer. Undefined values in the fields that are filtered will be matched by default to empty values for that field.
+
+It is recommended that consumers filter `fare_transfer_rules.txt` by the fields that define the characteristics of the transfer as outputted from trip planning software using `fare_leg_rules.txt`. The fields in `fare_transfer_rules.txt` that define characteristics of the transfer are:
+- `fare_transfer_rules.from_leg_group_id`
+- `fare_transfer_rules.to_leg_group_id`
+
+To process the cost of a multi-leg journey:
+1. Determine the applicable fares defined in `fare_leg_rules.txt` for all individual legs of travel.
+2. Process the actual cost of the journey based on the leg-to-leg transfers defined in `fare_transfer_rules.from_leg_group_id`, `fare_transfer_rules.to_leg_group_id`, and `fare_transfer_rules.fare_transfer_type`.
+
+|  Field Name | Type | Presence | Description |
+|  ------ | ------ | ------ | ------ |
+| `from_leg_group_id` | ID referencing `fare_leg_rules.leg_group_id` | Optional | Identifies a group of pre-transfer fare leg rules.<br><br>If there are no matching `fare_transfer_rules.from_leg_group_id` values to the `leg_group_id` being filtered, empty `fare_transfer_rules.from_leg_group_id` will be matched by default. |
+| `to_leg_group_id` | ID referencing `fare_leg_rules.leg_group_id` | Optional | Identifies a group of post-transfer fare leg rules.<br><br>If there are no matching `fare_transfer_rules.to_leg_group_id` values to the `leg_group_id` being filtered, empty `fare_transfer_rules.to_leg_group_id` will be matched by default. |
+| `is_symmetrical` | Enum | **Conditionally Required** | Indicates whether the fare transfer rule applies for transfers from `fare_transfer_rules.from_leg_group_id` to `fare_transfer_rules.to_leg_group_id` as well as from `fare_transfer_rules.to_leg_group_id` to `fare_transfer_rules.from_leg_group_id`.<br><br>Valid options are:<br>`0` - Not symmetrical.<br>`1` - Symmetrical. <br><br>Conditionally Required:<br>- **Required** if either `fare_leg_rules.from_leg_group_id` or `fare_leg_rules.to_leg_group_id` are defined.<br>- **Forbidden** otherwise.|
+| `spanning_limit` | Non-negative integer | **Conditionally Forbidden** | Defines how many consecutive legs the transfer rule may be applied to.<br><br>Valid options are:<br>`0` or empty - No limit.<br>`2` or more - Defines how many legs the transfer rule may span.<br><br>Conditionally Forbidden:<br>- **Forbidden** if `fare_transfer_rules.from_leg_group_id` does not equal `fare_transfer_rules.to_leg_group_id`.<br>- Optional if `fare_transfer_rules.from_leg_group_id` equals `fare_transfer_rules.to_leg_group_id`. |
```

Comment:
The case you described would likely be covered using the `duration_limit` fields in `fare_transfer_rules.txt`. I.e., unlimited transfers within X minutes.

For more complex transfer sequence rules, see `fare_transfer_rules.txt` fields `transfer_id` and `transfer_sequence`, as well as example S in the working doc: http://bit.ly/gtfs-fares. 

Since that feature is out of scope for this PR, please voice any feedback on that model in the doc!

Reply:
FWIW, walking between trips is an output of a trip planner, so it would not be considered a "leg" in pure GTFS terms. I believe the definition for "Leg" in this PR is clear:
> Travel in which a rider boards and alights between a pair of subsequent locations along a trip.